src/pykiso/pytest/plugin.py

```diff
@@ -0,0 +1,158 @@
+from __future__ import annotations
+
+import logging
+from functools import partial
+from pathlib import Path
+
+import pytest
+from _pytest.fixtures import FixtureDef, SubRequest
+from _pytest.logging import LoggingPlugin
+from _pytest.main import PytestPluginManager, Session
+from _pytest.unittest import TestCaseFunction
+
+from pykiso.config_parser import parse_config
+from pykiso.interfaces.dt_auxiliary import DTAuxiliaryInterface
+from pykiso.logging_initializer import initialize_logging
+from pykiso.test_coordinator.test_suite import flatten, tc_sort_key
+from pykiso.test_setup.config_registry import ConfigRegistry
+
+
+@pytest.hookimpl
+def pytest_auxiliary_start(aux: DTAuxiliaryInterface):
+    return aux.create_instance()
+
+
+@pytest.hookimpl
+def pytest_auxiliary_stop(aux: DTAuxiliaryInterface):
+    return aux.delete_instance()
+
+
+def pytest_addhooks(pluginmanager: PytestPluginManager):
+    from pykiso.pytest import hooks
+
+    pluginmanager.add_hookspecs(hooks)
+
+
+def add_auxiliary_fixture(
+    session: Session, aux_alias: str, aux_instance: DTAuxiliaryInterface
+):
+    def auxiliary_fixture(aux: DTAuxiliaryInterface = None):
+        pytest_auxiliary_start(aux)
+        yield aux
+        pytest_auxiliary_stop(aux)
+
+    # register the fixture
+    aux_func = partial(auxiliary_fixture, aux=aux_instance)
+    aux_func.__name__ = aux_alias
+    session._fixturemanager._arg2fixturedefs[aux_alias] = [
+        FixtureDef(
+            argname=aux_alias,
+            func=aux_func,
+            scope="session",
+            fixturemanager=session._fixturemanager,
+            baseid=None,
+            params=None,
+        ),
+    ]
+
+
+@pytest.hookimpl(trylast=True)
+def pytest_sessionstart(session: Session):
+    pytest_logger: LoggingPlugin = session.config.pluginmanager.get_plugin(
+        "logging-plugin"
+    )
+
+    initialize_logging(
+        log_path=None,
+        log_level=logging.getLevelName(pytest_logger.log_cli_level),
+        report_type="text",
+        verbose=True,
+    )
+    root_logger = logging.getLogger()
+
+    stream_handlers = [
+        hdlr for hdlr in root_logger.handlers if isinstance(hdlr, logging.StreamHandler)
+    ]
+    file_handlers = [
+        hdlr for hdlr in root_logger.handlers if isinstance(hdlr, logging.FileHandler)
+    ]
+
+    for hdlr in stream_handlers:
+        root_logger.removeHandler(hdlr)
+        root_logger.addHandler(pytest_logger.log_cli_handler)
+
+    for hdlr in file_handlers:
+        root_logger.removeHandler(hdlr)
+        root_logger.addHandler(pytest_logger.log_file_handler)
+
+
+def pytest_collection(session: Session):
+
```

Comment:
More comments would be helpful for someone like me that does not know pytest mechanism :)

Reply:
Totally agree, I'll add some :)

Reply:
The plugin is now split into multiple modules with more meaningful names and plenty of docstrings and comments ;)